content/security.md

```diff
@@ -0,0 +1,380 @@
+    fullName: { type: String, optional: true },
+    dateOfBirth: { type: Date, optional: true },
+  }),
+  run(fieldsToSet) {
+    Meteor.users.update(this.userId, {
+      $set: fieldsToSet
+    });
+  }
+});
+```
+
+The above method is great because you can have the flexibility of having some optional fields and only passing the ones you want to change. In particular, what makes it possible for this method is that the security considerations of setting one's full name and date of birth are the same - we don't have to do different security checks for different fields being set.
+
+#### Rate limiting: a first line of defense against brute force attacks
+
+Since Meteor methods can easily be called from anywhere - a malicious program, script in the browser console, etc - it is easy to fire many method calls in a very short amount of time. This means it's easy for an attacker to test lots of different inputs to find one that works. Meteor has built-in rate limiting for password login to stop password brute-forcing, but it's up to you to define rate limits for your other methods.
+
+In the Todos example app, we use the following code to set a basic rate limit on all methods:
+
+```js
+// Get list of all method names on Lists
+const LISTS_METHODS = _.pluck(Lists.methods, 'name');
+
+// Only allow 5 list operations per connection per second
+DDPRateLimiter.addRule({
+  name(name) {
+    return _.contains(LISTS_METHODS, name);
+  },
+
+  // Rate limit per connection ID
+  connectionId() { return true; }
+}, 5, 1000);
+```
+
+This will make every method only callable 5 times per second. This is a rate limit that shouldn't be noticeable by the user at all, but will prevent a malicious script from totally flooding the database with requests. You will need to tune the limit parameters to match your app's needs.
+
+Meteor's built-in rate limiter is useful in many situations; there are also community-built rate limiting packages that include additional features like reCAPTCHA integration, for example [`meteorhacks:sikka`](https://github.com/meteorhacks/sikka). XXX verify this is still a real thing
+
+XXX decided to skip method side effects because I can't come up with a good example, and audit-argument checks because mdg:method handles it
+
+### Publications
+
+Publications are the primary way clients retrieve data from a Meteor server. While with Methods the primary concern was making sure users can't modify the database in unexpected ways, with publications the main issue is filtering the data being returned so that a malicious user can't get access to data they aren't supposed to see.
+
+In a server-side-rendered framework like Ruby on Rails, it's sufficient to simply not display sensitive data in the returned HTML response. In Meteor, since the rendering is done on the client, an `if` statement in your HTML template is not secure; you need to do security at the data level to make sure that data is never sent in the first place.
```

Comment:
This is great. I think we need to call it out even stronger than this. Do we have method of emphasizing a sentence?